3rdparty/mbedtls/0001-Backport-2632-code-changes-to-2.16.patch

```diff
@@ -0,0 +1,505 @@
+                                                 MBEDTLS_ASN1_SEQUENCE ) );
+
+-    return mbedtls_x509write_crt_set_extension( ctx, MBEDTLS_OID_BASIC_CONSTRAINTS,
+-                                        MBEDTLS_OID_SIZE( MBEDTLS_OID_BASIC_CONSTRAINTS ),
+-                                        0, buf + sizeof(buf) - len, len );
++    return(
++        mbedtls_x509write_crt_set_extension( ctx, MBEDTLS_OID_BASIC_CONSTRAINTS,
++                             MBEDTLS_OID_SIZE( MBEDTLS_OID_BASIC_CONSTRAINTS ),
++                             0, buf + sizeof(buf) - len, len ) );
+ }
+
+ #if defined(MBEDTLS_SHA1_C)
+@@ -208,7 +217,8 @@ int mbedtls_x509write_crt_set_subject_key_identifier( mbedtls_x509write_cert *ct
+     size_t len = 0;
+
+     memset( buf, 0, sizeof(buf) );
+-    MBEDTLS_ASN1_CHK_ADD( len, mbedtls_pk_write_pubkey( &c, buf, ctx->subject_key ) );
++    MBEDTLS_ASN1_CHK_ADD( len,
++                mbedtls_pk_write_pubkey( &c, buf, ctx->subject_key ) );
+
+     ret = mbedtls_sha1_ret( buf + sizeof( buf ) - len, len,
+                             buf + sizeof( buf ) - 20 );
+@@ -218,11 +228,13 @@ int mbedtls_x509write_crt_set_subject_key_identifier( mbedtls_x509write_cert *ct
+     len = 20;
+
+     MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_len( &c, buf, len ) );
+-    MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_tag( &c, buf, MBEDTLS_ASN1_OCTET_STRING ) );
++    MBEDTLS_ASN1_CHK_ADD( len,
++            mbedtls_asn1_write_tag( &c, buf, MBEDTLS_ASN1_OCTET_STRING ) );
+
+-    return mbedtls_x509write_crt_set_extension( ctx, MBEDTLS_OID_SUBJECT_KEY_IDENTIFIER,
+-                                        MBEDTLS_OID_SIZE( MBEDTLS_OID_SUBJECT_KEY_IDENTIFIER ),
+-                                        0, buf + sizeof(buf) - len, len );
++    return mbedtls_x509write_crt_set_extension( ctx,
++                 MBEDTLS_OID_SUBJECT_KEY_IDENTIFIER,
++                 MBEDTLS_OID_SIZE( MBEDTLS_OID_SUBJECT_KEY_IDENTIFIER ),
++                 0, buf + sizeof(buf) - len, len );
+ }
+
+ int mbedtls_x509write_crt_set_authority_key_identifier( mbedtls_x509write_cert *ctx )
+@@ -233,7 +245,8 @@ int mbedtls_x509write_crt_set_authority_key_identifier( mbedtls_x509write_cert *
+     size_t len = 0;
+
+     memset( buf, 0, sizeof(buf) );
+-    MBEDTLS_ASN1_CHK_ADD( len, mbedtls_pk_write_pubkey( &c, buf, ctx->issuer_key ) );
++    MBEDTLS_ASN1_CHK_ADD( len,
++                          mbedtls_pk_write_pubkey( &c, buf, ctx->issuer_key ) );
+
+     ret = mbedtls_sha1_ret( buf + sizeof( buf ) - len, len,
+                             buf + sizeof( buf ) - 20 );
+@@ -243,15 +256,19 @@ int mbedtls_x509write_crt_set_authority_key_identifier( mbedtls_x509write_cert *
+     len = 20;
+
+     MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_len( &c, buf, len ) );
+-    MBEDTLS_ASN1_CHK_ADD( len, mbedtls_asn1_write_tag( &c, buf, MBEDTLS_ASN1_CONTEXT_SPECIFIC | 0 ) );
++    MBEDTLS_ASN1_CHK_ADD( len,
```

Comment:
Some of the diffs seem to be just code formatting. Can you please double-check if we are running clang-format incorrectly on 3rparty code.

Reply:
These are not clang-formatted to OE SDK project, these are the result of commit 6ad3fd1 in the upstream mbedtls project.